src/main/java/com/google/cloud/flink/bigquery/ProtobufUtils.java

```diff
@@ -0,0 +1,717 @@
+                  LegacySQLTypeName.GEOGRAPHY,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_BYTES)
+              .put(LegacySQLTypeName.TIME, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(LegacySQLTypeName.RECORD, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .build();
+
+  public static FlinkFnApi.Schema.FieldType toProtoType(LogicalType logicalType) {
+    return logicalType.accept(new ProtobufUtils.LogicalTypeToProtoTypeConverter());
+  }
+
+  public static ProtoSchema toProtoSchema(Schema schema) throws IllegalArgumentException {
+    try {
+      Descriptors.Descriptor descriptor = toDescriptor(schema);
+      return ProtoSchemaConverter.convert(descriptor);
+    } catch (Descriptors.DescriptorValidationException e) {
+      throw new IllegalArgumentException("Could not build Proto-Schema from Spark schema", e);
+    }
+  }
+
+  public static ProtoSchema toProtoSchema(RowType schema) throws IllegalArgumentException {
+    try {
+      Descriptors.Descriptor descriptor = toDescriptor(schema);
+      return ProtoSchemaConverter.convert(descriptor);
+    } catch (Descriptors.DescriptorValidationException e) {
+      throw new IllegalArgumentException("Could not build Proto-Schema from Spark schema", e);
+    }
+  }
+
+  public static ProtoSchema toProtoSchema(StructuredType schema) throws IllegalArgumentException {
+    try {
+      Descriptors.Descriptor descriptor = toDescriptor(schema);
+      return ProtoSchemaConverter.convert(descriptor);
+    } catch (Descriptors.DescriptorValidationException e) {
+      throw new IllegalArgumentException("Could not build Proto-Schema from Spark schema", e);
+    }
+  }
+
+  private static DescriptorProtos.FieldDescriptorProto.Type toProtoFieldType(
+      LegacySQLTypeName bqType) {
+    if (LegacySQLTypeName.RECORD.equals(bqType)) {
+      throw new IllegalStateException(
+          "Program attempted to return an atomic data-type for a RECORD");
+    }
+    return Preconditions.checkNotNull(
+        BigQueryToProtoType.get(bqType),
+        new IllegalArgumentException("Unexpected type: " + bqType.name()));
+  }
+
+  public static Descriptors.Descriptor toDescriptor(RowType schema)
+      throws Descriptors.DescriptorValidationException {
+    DescriptorProtos.DescriptorProto.Builder descriptorBuilder =
+        DescriptorProtos.DescriptorProto.newBuilder().setName("Schema");
+    List<Field> fieldList = new ArrayList<Field>();
+    Iterator<RowField> streamdata = schema.getFields().iterator();
+    while (streamdata.hasNext()) {
+      fieldList.addAll(getFields(streamdata.next(), null));
+    }
+    int initialDepth = 0;
+    DescriptorProtos.DescriptorProto descriptorProto =
+        buildDescriptorProtoWithFields(descriptorBuilder, FieldList.of(fieldList), initialDepth);
+
+    return createDescriptorFromProto(descriptorProto);
+  }
+
+  public static Descriptors.Descriptor toDescriptor(StructuredType schema)
```

Comment:
Are we using this method in PR #21? I do not see it utilized. I can only see `toDescriptor(RowType schema)` being used

Reply:
Removed methods which are not in scope at the moment